seml/config.py

```diff
@@ -268,14 +270,43 @@ def read_config(config_path):
         logging.error("Please specify a 'seml' dictionary in the experiment configuration.")
         sys.exit(1)
     seml_dict = config_dict['seml']
+    config_dir = os.path.abspath(os.path.dirname(config_path))
+    working_dir = config_dir
+    os.chdir(working_dir)
+
     del config_dict['seml']
+
     if "executable" not in seml_dict:
         logging.error("Please specify an executable path for the experiment.")
         sys.exit(1)
+    executable = seml_dict['executable']
     if "db_collection" in seml_dict:
         logging.warning("Specifying a the database collection in the config has been deprecated. "
                         "Please provide it via the command line instead.")
 
+    executable_relative_to_config = os.path.exists(executable)
+    executable_relative_to_project_root = False
+    if 'project_root_dir' in seml_dict:
+        working_dir = os.path.abspath(os.path.realpath(seml_dict['project_root_dir']))
+        seml_dict['has_root_dir'] = True
+        os.chdir(working_dir)  # use project root as base dir from now on
+        executable_relative_to_project_root = os.path.exists(executable)
+        del seml_dict['project_root_dir']  # from now on we use only the working dir
+    else:
+        seml_dict['has_root_dir'] = False
+        logging.warning("'project_root_dir' not defined in seml config. Source files will not be uploaded.")
+    seml_dict['working_dir'] = working_dir
+
+    if not (executable_relative_to_config or executable_relative_to_project_root):
+        logging.error(f"Could not find the executable.")
+        exit(1)
+    executable = os.path.abspath(executable)
+    seml_dict['executable'] = (str(Path(executable).relative_to(working_dir)) if executable_relative_to_project_root
+                               else str(Path(executable).relative_to(config_dir)))
+
+    if 'output_dir' in seml_dict:
+        seml_dict['output_dir'] = os.path.abspath(os.path.realpath(seml_dict['output_dir']))
```

Comment:
I think all of this logic for correctly setting the working_dir and executable deserves its own function now. :)